### PR TITLE
Feature/mapbox draw route

### PR DESCRIPTION
**Pull request by jakubprogramming**
This merge request functionality necessary to draw routes and switch them on button press.
Functionality that is not finished implementing or not needed any longer has been deactivated/removed.

### CODE REVIEW

File: ios/Runner.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dephiloper) [inline ios/Runner.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: please add this file to gitignore
already resolved by merging Dev into this branch.

File: lib/components/hikingmapbox.dart [hunk 2]
```diff
@@ -71,6 +95,133 @@ class _MapWidgetState extends State<MapWidget> {
...
+
+  void updateRoute() async {
+    List<LatLng> remainingRoute = [];
+    List<LatLng> passedRoute = _passedRoute;
+
+    for (int i = 0; i < _route.length; i++) {
+      LatLng latLng = _route[i];
+      double distanceToCurrentLocation =
+          OsmData.getDistance(latLng, _currentDeviceLocation);
+      //print("DISTANCE $distanceToCurrentLocation");
```
Comment (dephiloper) [inline lib/components/hikingmapbox.dart]: remove comment

File: lib/components/hikingmapbox.dart [hunk 3]
```diff
@@ -71,6 +95,133 @@ class _MapWidgetState extends State<MapWidget> {
...
+      if (distanceToCurrentLocation > 0.0002) {
+        remainingRoute.add(latLng);
+      } else {
+        passedRoute.add(latLng);
+        //print("i=" + i.toString() + " latlng = " + latLng.toString() + " length = " + passedRoute.length.toString());
```
Comment (dephiloper) [inline lib/components/hikingmapbox.dart]: remove comment

File: lib/components/hikingmapbox.dart (hunk 2 above)
Comment (dephiloper) [reply lib/components/hikingmapbox.dart]: remove print

File: lib/components/hikingmapbox.dart [hunk 4]
```diff
@@ -71,6 +95,133 @@ class _MapWidgetState extends State<MapWidget> {
...
+    await mapController.updateLine(_linePassedRoute, optionsPassedRoute);
+
+    setState(() {
+      _route = remainingRoute;
+      _passedRoute = passedRoute;
+    });
+  }
+
+  void updateRouteManually() async {
+    int numberOfNodesToUpdate = _route.length > 5 ? 5 : _route.length;
```
Comment (dephiloper) [inline lib/components/hikingmapbox.dart]: introduce const for 5

File: lib/services/osmdata.dart [hunk 5]
```diff
@@ -277,32 +305,80 @@ class OsmData{
...
-    result.forEach((edge) => resultNodes.addAll(graph.edgeToNodes(edge)));
+  List<List<Node>> _calculateRoundTripsWithoutPois(int alternativeRouteCount, double startLat, double startLong, double distanceInMeter) {
+    var randomGenerator = Random(1);
+    List<List<Node>> result = List();
+    for(var i =0; i<alternativeRouteCount; i++){
+      var initialHeading = randomGenerator.nextInt(360).floorToDouble();
+      var pointB = projectCoordinate(startLat, startLong, distanceInMeter/3, initialHeading);
+      var pointC = projectCoordinate(startLat, startLong, distanceInMeter/3, initialHeading + 60);
+
+    //      print("Initial Heading: " + initialHeading.toString());
```
Comment (dephiloper) [inline lib/services/osmdata.dart]: remove comments

File: pubspec.lock [hunk 6]
```diff
@@ -111,6 +111,13 @@ packages:
       url: "https://pub.dartlang.org"
```
Comment (dephiloper) [inline pubspec.lock:111]: ignore this

File: lib/components/hikingmapbox.dart (hunk 3 above)
Comment (dephiloper) [inline lib/components/hikingmapbox.dart]: remove comment

File: lib/components/hikingmapbox.dart [hunk 7]
```diff
@@ -50,12 +65,21 @@ class _MapWidgetState extends State<MapWidget> {
...
   initState() {
     super.initState();
-    _loadOfflineTiles();
+
+    _loadOfflineTiles().then((result) => {
+          requestLocationPermissionIfNotAlreadyGranted().then((result) {
+            getCurrentLocation().then((location) {
+              initRoutes();
+            });
+            updateCurrentLocationOnChange();
```
Comment (dephiloper) [inline lib/components/hikingmapbox.dart]: when the map is created you can receive a callback where you start tasks like:
```
getCurrentLocation()
initRoutes()
updateCurrentLocationOnChange()
```

This callback is already implemented see line 445 in the file onMapCreated.

File: android/build.gradle [hunk 8]
```diff
@@ -30,4 +30,4 @@ subprojects {
 
 task clean(type: Delete) {
     delete rootProject.buildDir
-}
+}
```
Comment (dephiloper) [inline android/build.gradle:33]: i think this should be reverted

**Review by dephiloper — CHANGES_REQUESTED** [review]
Please merge the current dev in this branch. Remove the unused comments and resolve the conflicts if you can comprehend them and are of the opinion that they make sense. Especially loading the route after the map was successfully created.

**Comment by jakubprogramming** [comment]
Ok is done

**Review by dephiloper — COMMENTED** [review]
done

**Pull request merged**